From 08f75643d6a4dd657ba488878f356d76a1db7c63 Mon Sep 17 00:00:00 2001
From: Hoang Nguyen <folliekazetani@protonmail.com>
Date: Thu, 7 Aug 2025 10:15:19 +0700
Subject: [PATCH] cmd/cue/cmd: use ParseInt with 64 bit size in cmd_after
 testdata

The current milliseconds since Epoch is out of bound for a 32-bit int. Use
int64 explicitly here. This change allows the test case to run successfully
on 32-bit platforms.

Closes #4021 as merged as of commit 13375f1c.

Note from Daniel: we didn't catch this in CI as we only run 32-bit tests
with `go test -short ./...`, and tests like this one which use sleeps
are marked as long. So we got a bit unlucky.

If this happens again in the future, we can teach our CI to skip the use
of `-short` when testing on the main branch.

Signed-off-by: Hoang Nguyen <folliekazetani@protonmail.com>
Change-Id: I2c679a6490469606f8be10706127015bea332b19
Reviewed-on: https://review.gerrithub.io/c/cue-lang/cue/+/1220033
Reviewed-by: Roger Peppe <rogpeppe@gmail.com>
Unity-Result: CUE porcuepine <cue.porcuepine@gmail.com>
TryBot-Result: CUEcueckoo <cueckoo@cuelang.org>
Patch-Source: https://review.gerrithub.io/c/cue-lang/cue/+/1220033
---
 cmd/cue/cmd/testdata/script/cmd_after.txtar | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
diff --git a/cmd/cue/cmd/testdata/script/cmd_after.txtar b/cmd/cue/cmd/testdata/script/cmd_after.txtar
index d05431d114e..1176ce297ab 100644
--- a/cmd/cue/cmd/testdata/script/cmd_after.txtar
+++ b/cmd/cue/cmd/testdata/script/cmd_after.txtar
@@ -38,8 +38,8 @@ command: after: {
 	}
 	t3: cli.Print & {
 		text: strconv.FormatBool(
-			strconv.Atoi(strings.TrimSpace(group.t1.stdout)) <
-				strconv.Atoi(strings.TrimSpace(group.t2.stdout))
+			strconv.ParseInt(strings.TrimSpace(group.t1.stdout), 10, 64) <
+				strconv.ParseInt(strings.TrimSpace(group.t2.stdout), 10, 64)
 			)
 	}
 	t4: cli.Print & {
